src/averageAges.js

```diff
@@ -15,7 +15,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
+  const mens = people.filter(person => person.sex === 'm');
+
+  const ages = century
+    ? [...mens]
+      .filter(men => Math.ceil(men.died / 100) === century)
+      .map(men => men.died - men.born)
+    : [...mens].map(men => men.died - men.born);
```

Comment:
No need to create copy of array here
```suggestion
  const ages = century
    ? mens.filter(men => Math.ceil(men.died / 100) === century)
      .map(men => men.died - men.born)
    : mens.map(men => men.died - men.born);
```